File: src/main/java/kitchenpos/tobe/product/domain/ProductNameValidator.java
package kitchenpos.tobe.product.domain;

import kitchenpos.tobe.product.domain.exception.InvalidProductNameException;
import org.springframework.stereotype.Component;
import org.springframework.util.StringUtils;

@Component
public class ProductNameValidator {

    private final PurgomalumAgent purgomalumAgent;

    public ProductNameValidator(PurgomalumAgent purgomalumAgent) {
        this.purgomalumAgent = purgomalumAgent;
    }

    public void validate(final String name) {
        if (!StringUtils.hasText(name)) {
            throw new InvalidProductNameException("상품명은 필수입니다.");
        }
        if (purgomalumAgent.containsProfanity(name)) {
            throw new InvalidProductNameException("상품명은 비속어를 포함하지 않아야 합니다.");
        }
    }

}

File: src/main/java/kitchenpos/products/tobe/application/ProductNameValidator.kt
package kitchenpos.products.tobe.application

import kitchenpos.products.infra.PurgomalumClient
import org.springframework.stereotype.Component

@Component
class ProductNameValidator(
    private val purgomalumClient: PurgomalumClient,
) {
    fun requireNormalName(
        name: String?,
    ) = require(name != null && isNotContainProfanity(name))

    private fun isNotContainProfanity(
        name: String,
    ) = !(purgomalumClient.containsProfanity(name))
}